util/sys/linux/sys_linux.go

```diff
@@ -51,3 +51,14 @@ func SetAffinity(cpus []int) error {
 	}
 	return unix.SchedSetaffinity(unix.Getpid(), &cpuSet)
 }
+
+// GetTargetDirectoryCapacity get the capacity (bytes) of directory
+func GetTargetDirectoryCapacity(path string) (int64, error) {
+	var stat syscall.Statfs_t
+	err := syscall.Statfs(path, &stat)
```

Comment:
I think this function is only unsupported on the WASM platform.
This function can be put under `util/sys/` package, and support both POSIX platform and windows platform, see https://stackoverflow.com/questions/20108520/get-amount-of-free-disk-space-using-go

Reply:
And better to add a test case.